sessionctx/variable/session.go

```diff
@@ -1284,6 +1284,84 @@ type SessionVars struct {
 	// OptPrefixIndexSingleScan indicates whether to do some optimizations to avoid double scan for prefix index.
 	// When set to true, `col is (not) null`(`col` is index prefix column) is regarded as index filter rather than table filter.
 	OptPrefixIndexSingleScan bool
+
+	// chunk alloc max cache chunk num
+	MaxReuseChunk int
+	// chunk alloc max cache column num
+	MaxReuseColumn int
+	// ChunkPool Several chunks and columns are cached
+	ChunkPool struct {
+		Lock  sync.Mutex
+		Alloc chunk.Allocator
+	}
+	// EnableReuseCheck indicates  request chunk whether use chunk alloc
+	EnableReuseCheck bool
+
+	// preuseChunkAlloc indicates whether pre statement use chunk alloc
+	// like select @@last_sql_use_alloc
+	preUseChunkAlloc bool
```

Comment:
Does it mean that the previous statement uses chunk buffer allocation? When would this variable be used, could you give some examples or situations?

Reply:
CREATE TABLE `t1` (
  `id` int(11) NOT NULL,
  `id2` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ;
insert into t1 values (1,1);
MySQL [test]> select * from t1 where id2 = 2;
+----+------+
| id | id2  |
+----+------+
|  2 |    2 |
+----+------+
1 row in set (0.00 sec)

MySQL [test]> select @@Last_sql_use_alloc;
+----------------------+
| @@Last_sql_use_alloc |
+----------------------+
|                    1 |
+----------------------+
1 row in set (0.00 sec)

MySQL [test]> select * from t1 where id = 1;
+----+------+
| id | id2  |
+----+------+
|  1 |    1 |
+----+------+
1 row in set (0.00 sec)

MySQL [test]> select @@Last_sql_use_alloc;
+----------------------+
| @@Last_sql_use_alloc |
+----------------------+
|                    0 |
+----------------------+
1 row in set (0.00 sec)

Reply:
added ut test